src/minio/core/policy_creator.py

```diff
@@ -0,0 +1,147 @@
+    PolicyPermissionLevel,
+    PolicySectionType,
+    PolicyStatement,
+    PolicyType,
+)
+
+logger = logging.getLogger(__name__)
+
+# Global system resource configuration
+SYSTEM_RESOURCE_CONFIG = {
+    # Spark-logs-related resources
+    # Creates policy with access to s3a://cdm-spark-job-logs/spark-job-logs/username/
+    "spark": {
+        "bucket": "cdm-spark-job-logs",
+        "base_prefix": "spark-job-logs",
+        "user_scoped": True,  # Whether this resource is user-specific
+        "permission_level": PolicyPermissionLevel.WRITE,
+    },
+    # Future system resources can be added here:
+    #
+    # Creates policy with access to s3a://cdm-task-service/task-service/
+    # "task_service": {
+    #     "bucket": "cdm-task-service",
+    #     "base_prefix": "task-service",
+    #     "user_scoped": False,
+    #     "permission_level": PolicyPermissionLevel.READ,
+    # },
+    #
+    # Creates policy with access to s3a://cdm-user-scratch/
+    # "scratch": {
+    #     "bucket": "cdm-user-scratch",
+    #     "base_prefix": "",
+    #     "user_scoped": False,
+    #     "permission_level": PolicyPermissionLevel.WRITE,
+    # },
+}
+
+
+class PolicyCreator:
+    """
+    Creator for building new MinIO policies from scratch.
+    """
+
+    def __init__(
+        self,
+        policy_type: PolicyType,
+        target_name: str,
+        config: MinIOConfig,
+    ):
+        """
+        Initialize the PolicyCreator for building a new policy from scratch.
+
+        Args:
+            policy_type: Type of policy being created (USER_HOME, USER_SYSTEM, GROUP_HOME)
+            target_name: Name of the target (username or group name)
+            config: MinIO configuration for bucket and path information
+        """
+
+        self.policy_type = policy_type
+        self.target_name = target_name
+        self.system_config = SYSTEM_RESOURCE_CONFIG
+
+        self.config = config
+        # user warehouse for spark tables
+        self.user_sql_warehouse_path = f"s3a://{self.config.default_bucket}/{self.config.users_sql_warehouse_prefix}/{self.target_name}"
+        # user warehouse for general files
+        self.user_general_warehouse_path = f"s3a://{self.config.default_bucket}/{self.config.users_general_warehouse_prefix}/{self.target_name}"
+        # group warehouse for general files
+        self.group_general_warehouse_path = f"s3a://{self.config.default_bucket}/{self.config.groups_general_warehouse_prefix}/{self.target_name}"
+
+        # Internal section management with Dict[PolicySectionType, List[PolicyStatement]]
+        self._sections: Dict[PolicySectionType, List[PolicyStatement]] = {
+            PolicySectionType.GLOBAL_PERMISSIONS: [],
+            PolicySectionType.BUCKET_ACCESS: [],
+            PolicySectionType.READ_PERMISSIONS: [],
+            PolicySectionType.WRITE_PERMISSIONS: [],
+            PolicySectionType.DELETE_PERMISSIONS: [],
+        }
+
+    def _get_user_system_paths(
```

Comment:
This appears to be unused - is there something missing from the PR?

Reply:
no - It will be used in the next PR. I thought it's related to the `SYSTEM_RESOURCE_CONFIG`, so I included it with this PR.